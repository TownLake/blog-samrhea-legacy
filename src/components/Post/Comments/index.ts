export { default as Comments } from "./Comments";
