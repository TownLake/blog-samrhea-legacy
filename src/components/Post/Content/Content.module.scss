@import "../../../assets/scss/variables";
@import "../../../assets/scss/mixins";

.content {
  @include margin-auto();
  max-width: $layout-post-single-width;
  @include padding(0, 0.5);

  .title {
    font-size: $typographic-base-font-size * 2;
    font-weight: 600;
    margin-left: auto;
    margin-right: auto;
    max-width: $layout-post-width;
    text-align: center;
    @include line-height(1.65);
    @include margin-top(1);
    @include margin-bottom(0);
  }

  .body {
    figure {
      @include margin-bottom(1);

      blockquote {
        font-style: italic;
        text-align: center;
        @include margin-top(0);
        @include padding(1, 0);

        p {
          font-size: $typographic-base-font-size * 1.6817;
          max-width: $layout-post-width;
          @include margin-top(0);
          @include margin-bottom(1);
          @include line-height(1.5);
        }
      }
    }

    a {
      text-decoration: underline;
    }

    * {
      margin-left: auto;
      margin-right: auto;
      max-width: $layout-post-width;
    }

    h2 > a {
      visibility: hidden;

      & > svg {
        fill: var(--color-typographic-base-font);
      }
    }

    img {
      max-width: 100%;
    }
  }
}

@include breakpoint-md {
  .content {
    @include padding-equal(0);

    .title {
      font-size: $typographic-base-font-size * 3;
      @include line-height(2.25);
      @include margin-top(2.25);
      @include margin-bottom(1.5);
    }

    .body {
      font-size: $typographic-base-font-size * 1.125;
      @include line-height(1.125);
      @include margin-bottom(1.125);

      p {
        font-size: $typographic-base-font-size * 1.125;
        @include line-height(1.125);
        @include margin-bottom(1.125);
      }

      h2 > a {
        visibility: unset;
        @include padding-right(0.5);
      }
    }
  }
}
