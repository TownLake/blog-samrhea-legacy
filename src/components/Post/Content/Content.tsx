import React from "react";

import * as styles from "./Content.module.scss";

interface Props {
  title: string;
  body: string;
}

const Content: React.FC<Props> = ({ body, title }: Props) => (
  <div className={styles.content}>
    <h1 className={styles.title}>{title}</h1>
    <div className={styles.body} dangerouslySetInnerHTML={{ __html: body }} />
  </div>
);

export default Content;
