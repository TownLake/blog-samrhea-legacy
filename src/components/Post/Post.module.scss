@import "src/assets/scss/variables";
@import "src/assets/scss/mixins";

.post {
  .content {
    @include margin-auto();
  }

  .footer {
    max-width: $layout-post-width;
    @include padding(0, 0.5);
    @include margin-auto();
  }

  .comments {
    max-width: $layout-post-width;
    @include padding(0, 0.5);
    @include margin-auto();
  }

  .buttons {
    align-items: center;
    display: flex;
    justify-content: center;
    text-align: center;
    @include margin(1, 0, 0, 0);

    .buttonArticles {
      @include margin(0, 0.5, 0, 0);
    }
  }
}

@include breakpoint-md {
  .post {
    .footer {
      @include padding-equal(0);
    }

    .comments {
      @include padding-equal(0);
    }

    .buttons {
      left: 30px;
      max-width: none;
      position: fixed;
      top: 30px;
      @include margin(0, 0.5, 0, 0);
    }
  }
}
