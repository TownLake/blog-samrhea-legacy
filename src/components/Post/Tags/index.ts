export { default as Tags } from "./Tags";
