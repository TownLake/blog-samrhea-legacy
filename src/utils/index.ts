export { default as getIcon } from "./get-icon";
export { default as getContactHref } from "./get-contact-href";
export { default as toKebabCase } from "./to-kebab-case";
export { default as getDefaultColorMode } from "./get-default-color-mode";
export * as testUtils from "./test-utils";
