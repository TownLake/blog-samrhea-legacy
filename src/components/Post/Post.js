// @flow strict
import React from 'react';
import { Link } from 'gatsby';
import Author from './Author';
import Comments from './Comments';
import Content from './Content';
import Meta from './Meta';
import Tags from './Tags';
import styles from './Post.module.scss';
import type { Node } from '../../types';

type Props = {
  post: Node
};

const Post = ({ post }: Props) => {
  const { html } = post;
  const { tagSlugs, slug } = post.fields;
  const { tags, title, date } = post.frontmatter;

  return (
    <div className={styles['post']}>
      <Link className={styles['post__home-button']} to="/">Home</Link>

      <div className={styles['post__content']}>
        <Content body={html} title={title} />
      </div>

      <div className={styles['post__footer']}>
        <Meta date={date} />
        {tags && tagSlugs && <Tags tags={tags} tagSlugs={tagSlugs} />}
        <Author />
      </div>

      <div className={styles['post__comments']}>
        <Comments postSlug={slug} postTitle={post.frontmatter.title} />
      </div>
    </div>
  );
};

export default Post;
