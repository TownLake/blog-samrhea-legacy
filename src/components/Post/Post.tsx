import React from "react";

import { Button } from "@/components/Button";
import { ThemeSwitcher } from "@/components/ThemeSwitcher";
import type { Node } from "@/types";

import { Author } from "./Author";
import { Comments } from "./Comments";
import { Content } from "./Content";
import { Meta } from "./Meta";
import { Tags } from "./Tags";

import * as styles from "./Post.module.scss";

interface Props {
  post: Node;
}

const Post: React.FC<Props> = ({ post }: Props) => {
  const { html } = post;
  const { tagSlugs, slug } = post.fields;
  const { tags, title, date } = post.frontmatter;

  return (
    <div className={styles.post}>
      <div className={styles.buttons}>
        <Button className={styles.buttonArticles} title="All Articles" to="/" />
        <ThemeSwitcher />
      </div>

      <div className={styles.content}>
        <Content body={html} title={title} />
      </div>

      <div className={styles.footer}>
        <Meta date={date} />
        {tags && tagSlugs && <Tags tags={tags} tagSlugs={tagSlugs} />}
        <Author />
      </div>

      <div className={styles.comments}>
        <Comments postSlug={slug} postTitle={post.frontmatter.title} />
      </div>
    </div>
  );
};

export default Post;
