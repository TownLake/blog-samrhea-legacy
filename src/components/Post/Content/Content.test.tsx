import React from "react";

import { Content } from "@/components/Post/Content";
import * as mocks from "@/mocks";
import { testUtils } from "@/utils";

describe("Content", () => {
  test("renders correctly", () => {
    const props = {
      title: mocks.markdownRemark.frontmatter.title,
      body: mocks.markdownRemark.html,
    };

    const tree = testUtils
      .createSnapshotsRenderer(<Content {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
