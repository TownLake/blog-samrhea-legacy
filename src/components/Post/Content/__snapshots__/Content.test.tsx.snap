// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders correctly 1`] = `
<div>
  <h1>
    Perfecting the Art of Perfection
  </h1>
  <div
    dangerouslySetInnerHTML={
      {
        "__html": "<p>An Essay on Typography by Eric Gill takes the reader back to the year 1930. The year when a conflict between two worlds came to its term. The machines of the industrial world finally took over the handicrafts.</p>",
      }
    }
  />
</div>
`;
